mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp

```diff
@@ -26,6 +26,8 @@ llvm::StringRef mlir::sycl::memoryAccessModeAsString(
     return "discard_read_write";
   case MemoryAccessMode::Atomic:
     return "atomic";
+  default:
```

Comment:
As the switch already covered all possible MemoryAccessMode
```
enum class MemoryAccessMode {
  Read = 1024,
  Write,
  ReadWrite,
  DiscardWrite,
  DiscardReadWrite,
  Atomic
};
```
would adding a default case add a warning when compiling with clang++?

Reply:
I recall it did in one of my past PRs, and newer version of gcc should warn as well AFAIK.

Reply:
I think clang will correctly recognize the llvm_unreachable and suppress the warnings.

Reply:
I tried to compile with clang++ and got:
```
llvm/mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp:28:3: warning: default label in switch which covers all enumeration values [-Wcovered-switch-default]
  default:
  ^
1 warning generated.
```

Reply:
@whitneywhtsang, please, try with last push

Reply:
I tried just for this switch, and no warnings with the change from the last push.